src/service/url-replacements-impl.js

```diff
@@ -146,6 +151,21 @@ export class GlobalVariableSource extends VariableSource {
       return Services.viewerForDoc(this.ampdoc).getReferrerUrl();
     }));
 
+    // Like DOCUMENT_REFERRER, but returns null if the referrer is same domain
+    // as the current page. CDN proxy is considered same domain.
+    this.setAsync('EXTERNAL_REFERRER', /** @type {AsyncResolverDef} */(() => {
+      return Services.viewerForDoc(this.ampdoc).getReferrerUrl()
+          .then(referrer => {
+            if (!referrer) {
+              return referrer;
+            }
+            const referrerHostname = parseUrl(getSourceUrl(referrer)).hostname;
```

Comment:
The feature is to set referrer to null if it was from CDN proxy of the same domain.